File: leetCode/two-pointers/language/python/is-palyndrome.py
def isPalyndrome(word):
    left = 0
    right = len(word) - 1
    while left < right:
        if word[left] != word[right]:
            return False
        left += 1
        right -= 1

    return True


if __name__ == "__main__":
    print(isPalyndrome("abba"))
    print(isPalyndrome("abbaa"))

File: leetCode/two-pointers/language/javascript/is-palyndrome.ts
function isPalyndrome(word: string): boolean {
  let left = 0;
  let right = word.length - 1;

  while (left < right) {
    if (word[left] !== word[right]) {
      return false;
    }

    left++;
    right--;
  }

  return true;
}

const test1 = isPalyndrome("abba");
console.log(test1);

const test2 = isPalyndrome("abbaa");
console.log(test2);
